### ISSUE TITLE
Not all build output is suppressed using `-Q -q`

### ISSUE THREAD

**Issue by michalrus** [NixOS/nix#1341]
Reproduction:

```shell
#!/bin/sh

cd $(mktemp -d)

touch project.cabal

echo '(import <nixpkgs> {}).haskellPackages.callCabal2nix "project" ./. {}' >default.nix

nix-build --quiet -Q
```

Output:

```
installing
builder for ‘/nix/store/v57j4jnbqlw41gbm1bmx3y3hv9manvy6--.drv’ failed with exit code 1
…
```

Expected:

```
builder for ‘/nix/store/v57j4jnbqlw41gbm1bmx3y3hv9manvy6--.drv’ failed with exit code 1
…
```

This surplus `installing` most probably comes from https://github.com/NixOS/nixpkgs/blob/7daec8a234da5b097544dd4fb3fa1d2012870d08/pkgs/stdenv/generic/setup.sh#L827.

Causes: https://github.com/michalrus/intero-nix-shim/issues/1.

/cc @shlevy

**Comment by vcunat** [comment]
Yes, `-Q` doesn't apply to anything needed during *evaluation* (including builds but we tend to avoid that in nixpkgs).  I don't know if that's intentional or not.

**Comment by michalrus** [comment]
Ah! It would be very useful to have _a_ flag (or that flag) to silence evaluation output, as well.

**Comment by shlevy** [comment]
Yeah,this is on my list to figure out and fix

**Comment by hraban** [comment]
This still seems relevant. I found this when googling for a way to make nix actually completely quiet.
